### PR TITLE
Search string drag and drop, and make fluid p2p bidirectional

### PR DESCRIPTION
**Pull request by repo-alt**
image
image

### CODE REVIEW

**Comment by repo-alt** [comment]
Unified that with https://github.com/GTNewHorizons/Applied-Energistics-2-Unofficial/pull/76
GTNewHorizons/GT-New-Horizons-Modpack#8891

File: src/main/java/appeng/parts/p2p/PartP2PLiquids.java [hunk 1]
```diff
@@ -254,10 +262,19 @@ public FluidStack drain( final ForgeDirection from, final int maxDrain, final bo
...
+			return false;
+		final IFluidHandler handler = l.getTarget();
+		return handler != null && handler.canFill(l.getSide().getOpposite(), input);
+	}
+
 	@Override
 	public boolean canFill( final ForgeDirection from, final Fluid fluid )
 	{
-		return !this.isOutput() && from == this.getSide() && !this.getOutputs( fluid ).isEmpty();
+		return from == this.getSide() && (!this.getOutputs( fluid ).isEmpty() || inputAcceptsFluid(fluid));
```
Comment (TheElan) [inline src/main/java/appeng/parts/p2p/PartP2PLiquids.java:277]: What do you think about splitting this into cars to clarify what you are testing?

File: src/main/java/appeng/integration/modules/NEIHelpers/NEIGuiHandler.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+public class NEIGuiHandler extends INEIGuiAdapter {
+    @Override
+    public boolean handleDragNDrop(GuiContainer gui, int mousex, int mousey, ItemStack draggedStack, int button) {
+        if (gui instanceof GuiMEMonitorable && draggedStack != null && draggedStack.getItem() != null)
+        {
+            GuiMEMonitorable gmm = (GuiMEMonitorable)gui;
+            if (gmm.isOverSearchField(mousex, mousey))
+            {
+                gmm.setSearchString(draggedStack.getDisplayName(), true);
```
Comment (TheElan) [inline src/main/java/appeng/integration/modules/NEIHelpers/NEIGuiHandler.java:16]: Would use suggest full words for a name, it's much more  readable then acronyms. You don't need to use full name, som thing like `monitorableUI` would be sufficient

File: src/main/java/appeng/integration/modules/NEIHelpers/NEIGuiHandler.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package appeng.integration.modules.NEIHelpers;
+
+import appeng.client.gui.implementations.GuiMEMonitorable;
+import codechicken.nei.api.INEIGuiAdapter;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.item.ItemStack;
+
+public class NEIGuiHandler extends INEIGuiAdapter {
```
Comment (TheElan) [inline src/main/java/appeng/integration/modules/NEIHelpers/NEIGuiHandler.java:8]: Love when somebody brings naming convention from other language and uses it inconsistently (`I` prefix means interface, in java world the practice is to append `Impl` suffux in worst case).

(Not saying you should do anything about this)

**Review by TheElan — APPROVED** [review]
Looks fine to me

**Pull request merged**